### PR TITLE
[High-Level-API] Update MNIST to use optimizer_func

### PR DESCRIPTION
**Pull request by jetfuel**

### CODE REVIEW

File: 02.recognize_digits/README.md [hunk 1]
```diff
@@ -146,7 +146,7 @@ Here are the quick overview on the major fluid API complements.
 This is where you specify the network flow.
 1. `train_program`: A function that specify how to get avg_cost from `inference_program` and labels.
 This is where you specify the loss calculations.
-1. `optimizer`: Configure how to minimize the loss. Paddle supports most major optimization methods.
+1. `optimizer_func`: Configure how to minimize the loss. Paddle supports most major optimization methods.
```
Comment (sidgoyal78) [inline 02.recognize_digits/README.md]: Can we elaborate a bit? I like the description of the `train_program`. Can we write something like "A function that specifies the configuration of the the optimizer. The optimizer is responsible for minimizing the loss and driving the training. Paddle supports many different optimizers."

File: 02.recognize_digits/README.md [hunk 2]
```diff
@@ -245,6 +245,15 @@ def train_program():
     return [avg_cost, acc]
 ```
 
+#### Optimizer Function Configuration
+
+In the following `Adam` optimizer, `learning_rate` means the speed at which the network training converges.
```
Comment (sidgoyal78) [inline 02.recognize_digits/README.md]: Can we rephrase: `learning_rate` specifies the learning rate in the optimization procedure.

**Review by sidgoyal78 — CHANGES_REQUESTED** [review]

File: 02.recognize_digits/README.md (hunk 2 above)
Comment (jetfuel) [reply 02.recognize_digits/README.md]: sure, sounds good.

**Review by sidgoyal78 — APPROVED** [review]
LGTM
